src/constants/algorithm.rs

```diff
@@ -254,23 +254,15 @@ impl TryFrom<TPM2_ALG_ID> for HashingAlgorithm {
 /// Enum containing signature schemes
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub enum SignatureScheme {
-    RsaSsa,
-    RsaPss,
-    EcDsa,
-    EcDaa,
-    EcSchnorr,
-    Sm2,
+    Rsa(RsaSignatureScheme),
+    Ecc(EccSignatureScheme),
 }
 
 impl From<SignatureScheme> for TPM2_ALG_ID {
     fn from(signature_scheme: SignatureScheme) -> Self {
         match signature_scheme {
-            SignatureScheme::RsaSsa => TPM2_ALG_RSASSA,
-            SignatureScheme::RsaPss => TPM2_ALG_RSAPSS,
-            SignatureScheme::EcDsa => TPM2_ALG_ECDSA,
-            SignatureScheme::EcDaa => TPM2_ALG_ECDAA,
-            SignatureScheme::EcSchnorr => TPM2_ALG_ECSCHNORR,
-            SignatureScheme::Sm2 => TPM2_ALG_SM2,
+            SignatureScheme::Rsa(ss) => ss.into(),
```

Comment:
Yes! This is much better. The previous(mine) was just taking any alg in the specification and putting up as enum without any closer thought behind it. This is a huge improvement,